sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,944 @@
+ * @return The result of the operation.
+ */
+// Open Q: do we want a SW version, as a convenience for programming the digest
+// of a SW partition?
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_hw_digest(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition);
+
+/**
+ * Gets the buffered digest value for the given partition.
+ *
+ * Note that this value is only updated when the device is reset; if the digest
+ * has not been computed yet, or has been computed but not since device reset,
+ * this function will return an error.
+ *
+ * The lifecycle partition does not have a digest and will result in an error
+ * being returned.
+ *
+ * @param otp An OTP handle.
+ * @param partition The parittion to get a digest for.
+ * @param[out] digest Out-param for the digest.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_digest_result_t dif_otp_ctrl_get_digest(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint64_t *digest);
+
+/**
+ * Gets the buffered lifecycle state.
+ *
+ * The returned value represents the lifecycle state of the device as of the
+ * most recent reset.
+ *
+ * @param otp An OTP handle.
+ * @param[out] state Out-param for the lifecycle state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_result_t dif_otp_ctrl_get_lifecycle_state(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_lifecycle_state_t *state);
+
+/**
+ * Performs a memory-mapped read of the given partition, if it supports them.
+ *
+ * In particular, this function will read `len` words, starting at `address`,
+ * relative to the start of `partition`.
+ *
+ * The same caveats for `dif_otp_ctrl_dai_read_start()` apply to `address`; in
+ * addition, `address + len` must also be in-range and must not overflow.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to read from.
+ * @param address A partition-releative address to read from.
+ * @param[out] buf A buffer of words to write read values to.
+ * @param len The number of words to read.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_read_polled(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint32_t address, uint32_t *buf, size_t len);
+
+/**
+ * Performs a memory-mapped read of the TEST region.
+ *
+ * In particular, this function will read `len` words, starting at `address`.
+ *
+ * The same caveats for `dif_otp_ctrl_dai_read_start()` apply to `address`; in
+ * addition, `address + len` must also be in-range and must not overflow.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to read from.
+ * @param[out] buf A buffer of words to write read values to.
+ * @param len The number of words to read.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_result_t dif_otp_ctrl_read_test(const dif_otp_ctrl_t *otp,
```

Comment:
I think eventually, this TEST interface will only need to support single word writes/reads, since it basically connects to a hidden CSR bank inside the proprietary OTP wrapper. The actual implementation is however on Nuvoton, so we do not exactly know yet how this looks like under the hood.

So for the time being, this should be fine.